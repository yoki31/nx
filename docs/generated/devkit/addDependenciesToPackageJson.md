# Function: addDependenciesToPackageJson

â–¸ **addDependenciesToPackageJson**(`tree`, `dependencies`, `devDependencies`, `packageJsonPath?`, `keepExistingVersions?`): [`GeneratorCallback`](../../devkit/documents/GeneratorCallback)

Add Dependencies and Dev Dependencies to package.json

For example:

```typescript
addDependenciesToPackageJson(tree, { react: 'latest' }, { jest: 'latest' });
```

This will **add** `react` and `jest` to the dependencies and devDependencies sections of package.json respectively.

#### Parameters

| Name                    | Type                                  | Description                                                                 |
| :---------------------- | :------------------------------------ | :-------------------------------------------------------------------------- |
| `tree`                  | [`Tree`](../../devkit/documents/Tree) | Tree representing file system to modify                                     |
| `dependencies`          | `Record`\<`string`, `string`\>        | Dependencies to be added to the dependencies section of package.json        |
| `devDependencies`       | `Record`\<`string`, `string`\>        | Dependencies to be added to the devDependencies section of package.json     |
| `packageJsonPath?`      | `string`                              | Path to package.json                                                        |
| `keepExistingVersions?` | `boolean`                             | If true, prevents existing dependencies from being bumped to newer versions |

#### Returns

[`GeneratorCallback`](../../devkit/documents/GeneratorCallback)

Callback to install dependencies only if necessary, no-op otherwise
