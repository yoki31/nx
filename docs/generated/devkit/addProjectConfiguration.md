# Function: addProjectConfiguration

â–¸ **addProjectConfiguration**(`tree`, `projectName`, `projectConfiguration`, `standalone?`): `void`

Adds project configuration to the Nx workspace.

#### Parameters

| Name                   | Type                                                                  | Default value | Description                                                             |
| :--------------------- | :-------------------------------------------------------------------- | :------------ | :---------------------------------------------------------------------- |
| `tree`                 | [`Tree`](../../devkit/documents/Tree)                                 | `undefined`   | the file system tree                                                    |
| `projectName`          | `string`                                                              | `undefined`   | unique name. Often directories are part of the name (e.g., mydir-mylib) |
| `projectConfiguration` | [`ProjectConfiguration`](../../devkit/documents/ProjectConfiguration) | `undefined`   | project configuration                                                   |
| `standalone`           | `boolean`                                                             | `true`        | whether the project is configured in workspace.json or not              |

#### Returns

`void`
