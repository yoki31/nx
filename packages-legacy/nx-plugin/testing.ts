export * from '@nx/plugin/testing';
