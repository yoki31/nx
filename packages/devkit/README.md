<p style="text-align: center;">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/nrwl/nx/master/images/nx-dark.svg">
    <img alt="Nx - Smart Monorepos · Fast CI" src="https://raw.githubusercontent.com/nrwl/nx/master/images/nx-light.svg" width="100%">
  </picture>
</p>

{{links}}

<hr>

# Nx: Smart Monorepos · Fast CI

Nx is a build system with built-in tooling and advanced CI capabilities. It helps you maintain and scale monorepos, both locally and on CI.

This package contains a set of utilities for creating Nx plugins.

{{content}}
