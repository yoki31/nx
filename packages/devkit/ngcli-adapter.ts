/**
 * @category Ng CLI Adapter
 */
// eslint-disable-next-line @typescript-eslint/no-restricted-imports
export {
  wrapAngularDevkitSchematic,
  mockSchematicsForTesting,
  NxScopedHost,
} from 'nx/src/adapter/ngcli-adapter';
