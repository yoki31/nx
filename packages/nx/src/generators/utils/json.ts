import type { Tree } from '../tree';
import { parseJson, serializeJson } from '../../utils/json';
import type { JsonParseOptions, JsonSerializeOptions } from '../../utils/json';

/**
 * Reads a json file, removes all comments and parses JSON.
 *
 * @param tree - file system tree
 * @param path - file path
 * @param options - Optional JSON Parse Options
 */
export function readJson<T extends object = any>(
  tree: Tree,
  path: string,
  options?: JsonParseOptions
): T {
  if (!tree.exists(path)) {
    throw new Error(`Cannot find ${path}`);
  }
  try {
    return parseJson(tree.read(path, 'utf-8'), options);
  } catch (e) {
    throw new Error(`Cannot parse ${path}: ${e.message}`);
  }
}

/**
 * Writes a JSON value to the file system tree

 * @param tree File system tree
 * @param path Path of JSON file in the Tree
 * @param value Serializable value to write
 * @param options Optional JSON Serialize Options
 */
export function writeJson<T extends object = object>(
  tree: Tree,
  path: string,
  value: T,
  options?: JsonSerializeOptions
): void {
  const serialized = serializeJson(value, options);
  tree.write(path, `${serialized}\n`);
}

/**
 * Updates a JSON value to the file system tree
 *
 * @param tree File system tree
 * @param path Path of JSON file in the Tree
 * @param updater Function that maps the current value of a JSON document to a new value to be written to the document
 * @param options Optional JSON Parse and Serialize Options
 */
export function updateJson<T extends object = any, U extends object = T>(
  tree: Tree,
  path: string,
  updater: (value: T) => U,
  options?: JsonParseOptions & JsonSerializeOptions
): void {
  const updatedValue = updater(readJson(tree, path, options));
  writeJson(tree, path, updatedValue, options);
}
