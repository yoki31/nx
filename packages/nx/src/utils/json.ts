import { parse, printParseErrorCode, stripComments } from 'jsonc-parser';
import type { ParseError, ParseOptions } from 'jsonc-parser';
import { LinesAndColumns } from 'lines-and-columns';
import { codeFrameColumns } from './code-frames';

export { stripComments as stripJsonComments };

export interface JsonParseOptions extends ParseOptions {
  /**
   * Expect JSON with javascript-style
   * @default false
   */
  expectComments?: boolean;
  /**
   * Disallow javascript-style
   * @default false
   */
  disallowComments?: boolean;

  /**
   * Allow trailing commas in the JSON content
   */
  allowTrailingComma?: boolean;
}

export interface JsonSerializeOptions {
  /**
   * the whitespaces to add as indentation to make the output more readable.
   * @default 2
   */
  spaces?: number;
}

/**
 * Parses the given JSON string and returns the object the JSON content represents.
 * By default javascript-style comments and trailing commas are allowed.
 *
 * @param input JSON content as string
 * @param options JSON parse options
 * @returns Object the JSON content represents
 */
export function parseJson<T extends object = any>(
  input: string,
  options?: JsonParseOptions
): T {
  try {
    if (options?.expectComments !== true) {
      return JSON.parse(input);
    }
  } catch {}

  options = { allowTrailingComma: true, ...options };

  const errors: ParseError[] = [];
  const result: T = parse(input, errors, options);

  if (errors.length > 0) {
    throw new Error(formatParseError(input, errors[0]));
  }

  return result;
}

/**
 * Nicely formats a JSON error with context
 *
 * @param input JSON content as string
 * @param parseError jsonc ParseError
 * @returns
 */
function formatParseError(input: string, parseError: ParseError) {
  const { error, offset, length } = parseError;
  let { line, column } = new LinesAndColumns(input).locationForIndex(offset);
  line++;
  column++;

  return (
    `${printParseErrorCode(error)} in JSON at ${line}:${column}\n` +
    codeFrameColumns(input, {
      start: { line, column },
      end: { line, column: column + length },
    }) +
    '\n'
  );
}

/**
 * Serializes the given data to a JSON string.
 * By default the JSON string is formatted with a 2 space indentation to be easy readable.
 *
 * @param input Object which should be serialized to JSON
 * @param options JSON serialize options
 * @returns the formatted JSON representation of the object
 */
export function serializeJson<T extends object = object>(
  input: T,
  options?: JsonSerializeOptions
): string {
  return JSON.stringify(input, null, options?.spaces ?? 2);
}
