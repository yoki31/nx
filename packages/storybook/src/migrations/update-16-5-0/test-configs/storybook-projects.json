{
  "my-custom-place-nested-app": {
    "name": "my-custom-place-nested-app",
    "$schema": "../../../../../node_modules/nx/schemas/project-schema.json",
    "sourceRoot": "apps/my/custom/place/nested-app/src",
    "projectType": "application",
    "targets": {
      "build": {
        "dependsOn": ["^build"],
        "inputs": ["production", "^production"],
        "executor": "@nx/vite:build",
        "outputs": ["{options.outputPath}"],
        "defaultConfiguration": "production",
        "options": {
          "outputPath": "dist/apps/my/custom/place/nested-app",
          "mode": "production"
        },
        "configurations": {
          "development": { "mode": "development" },
          "production": { "mode": "production" }
        }
      },
      "serve": {
        "executor": "@nx/vite:dev-server",
        "defaultConfiguration": "development",
        "options": { "buildTarget": "my-custom-place-nested-app:build" },
        "configurations": {
          "development": {
            "buildTarget": "my-custom-place-nested-app:build:development",
            "hmr": true
          },
          "production": {
            "buildTarget": "my-custom-place-nested-app:build:production",
            "hmr": false
          }
        }
      },
      "preview": {
        "executor": "@nx/vite:preview-server",
        "defaultConfiguration": "development",
        "options": { "buildTarget": "my-custom-place-nested-app:build" },
        "configurations": {
          "development": {
            "buildTarget": "my-custom-place-nested-app:build:development"
          },
          "production": {
            "buildTarget": "my-custom-place-nested-app:build:production"
          }
        }
      },
      "test": {
        "inputs": ["default", "^production", "{workspaceRoot}/jest.preset.js"],
        "executor": "@nx/vite:test",
        "outputs": ["coverage/apps/my/custom/place/nested-app"],
        "options": {
          "passWithNoTests": true,
          "reportsDirectory": "../../../../../coverage/apps/my/custom/place/nested-app"
        },
        "configurations": {}
      },
      "lint": {
        "inputs": [
          "default",
          "{workspaceRoot}/.eslintrc.json",
          "{workspaceRoot}/.eslintignore"
        ],
        "executor": "@nx/linter:eslint",
        "outputs": ["{options.outputFile}"],
        "options": {
          "lintFilePatterns": [
            "apps/my/custom/place/nested-app/**/*.{ts,tsx,js,jsx}"
          ]
        },
        "configurations": {}
      },
      "serve-static": {
        "executor": "@nx/web:file-server",
        "options": { "buildTarget": "my-custom-place-nested-app:build" }
      },
      "storybook": {
        "executor": "@nx/storybook:storybook",
        "options": {
          "port": 4400,
          "configDir": "apps/my/custom/place/nested-app/.storybook"
        },
        "configurations": { "ci": { "quiet": true } }
      },
      "build-storybook": {
        "inputs": ["default", "^production", "!{projectRoot}/.storybook/**/*"],
        "executor": "@nx/storybook:build",
        "outputs": ["{options.outputDir}"],
        "options": {
          "outputDir": "dist/storybook/my-custom-place-nested-app",
          "configDir": "apps/my/custom/place/nested-app/.storybook"
        },
        "configurations": { "ci": { "quiet": true } }
      },
      "static-storybook": {
        "executor": "@nx/web:file-server",
        "options": {
          "buildTarget": "my-custom-place-nested-app:build-storybook",
          "staticFilePath": "dist/storybook/my-custom-place-nested-app"
        },
        "configurations": {
          "ci": {
            "buildTarget": "my-custom-place-nested-app:build-storybook:ci"
          }
        }
      }
    },
    "tags": [],
    "root": "apps/my/custom/place/nested-app",
    "implicitDependencies": []
  },
  "webpack-app": {
    "name": "webpack-app",
    "$schema": "../../node_modules/nx/schemas/project-schema.json",
    "sourceRoot": "apps/webpack-app/src",
    "projectType": "application",
    "targets": {
      "build": {
        "dependsOn": ["^build"],
        "inputs": ["production", "^production"],
        "executor": "@nx/webpack:webpack",
        "outputs": ["{options.outputPath}"],
        "defaultConfiguration": "production",
        "options": {
          "compiler": "babel",
          "outputPath": "dist/apps/webpack-app",
          "index": "apps/webpack-app/src/index.html",
          "baseHref": "/",
          "main": "apps/webpack-app/src/main.tsx",
          "tsConfig": "apps/webpack-app/tsconfig.app.json",
          "assets": [
            "apps/webpack-app/src/favicon.ico",
            "apps/webpack-app/src/assets"
          ],
          "styles": ["apps/webpack-app/src/styles.css"],
          "scripts": [],
          "isolatedConfig": true,
          "webpackConfig": "apps/webpack-app/webpack.config.js",
          "fileReplacements": [
            {
              "replace": "apps/webpack-app/src/environments/environment.ts",
              "with": "apps/webpack-app/src/environments/environment.prod.ts"
            }
          ],
          "optimization": true,
          "outputHashing": "all",
          "sourceMap": false,
          "namedChunks": false,
          "extractLicenses": true,
          "vendorChunk": false
        },
        "configurations": {
          "development": {
            "extractLicenses": false,
            "optimization": false,
            "sourceMap": true,
            "vendorChunk": true
          },
          "production": {
            "fileReplacements": [
              {
                "replace": "apps/webpack-app/src/environments/environment.ts",
                "with": "apps/webpack-app/src/environments/environment.prod.ts"
              }
            ],
            "optimization": true,
            "outputHashing": "all",
            "sourceMap": false,
            "namedChunks": false,
            "extractLicenses": true,
            "vendorChunk": false
          }
        }
      },
      "serve": {
        "executor": "@nx/webpack:dev-server",
        "defaultConfiguration": "development",
        "options": { "buildTarget": "webpack-app:build", "hmr": true },
        "configurations": {
          "development": { "buildTarget": "webpack-app:build:development" },
          "production": {
            "buildTarget": "webpack-app:build:production",
            "hmr": false
          }
        }
      },
      "lint": {
        "inputs": [
          "default",
          "{workspaceRoot}/.eslintrc.json",
          "{workspaceRoot}/.eslintignore"
        ],
        "executor": "@nx/linter:eslint",
        "outputs": ["{options.outputFile}"],
        "options": {
          "lintFilePatterns": ["apps/webpack-app/**/*.{ts,tsx,js,jsx}"]
        },
        "configurations": {}
      },
      "serve-static": {
        "executor": "@nx/web:file-server",
        "options": { "buildTarget": "webpack-app:build" }
      },
      "test": {
        "inputs": ["default", "^production", "{workspaceRoot}/jest.preset.js"],
        "executor": "@nx/jest:jest",
        "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
        "options": {
          "jestConfig": "apps/webpack-app/jest.config.ts",
          "passWithNoTests": true
        },
        "configurations": { "ci": { "ci": true, "codeCoverage": true } }
      },
      "storybook": {
        "executor": "@nx/storybook:storybook",
        "options": { "port": 4400, "configDir": "apps/webpack-app/.storybook" },
        "configurations": { "ci": { "quiet": true } }
      },
      "build-storybook": {
        "inputs": ["default", "^production", "!{projectRoot}/.storybook/**/*"],
        "executor": "@nx/storybook:build",
        "outputs": ["{options.outputDir}"],
        "options": {
          "outputDir": "dist/storybook/webpack-app",
          "configDir": "apps/webpack-app/.storybook"
        },
        "configurations": { "ci": { "quiet": true } }
      },
      "static-storybook": {
        "executor": "@nx/web:file-server",
        "options": {
          "buildTarget": "webpack-app:build-storybook",
          "staticFilePath": "dist/storybook/webpack-app"
        },
        "configurations": {
          "ci": { "buildTarget": "webpack-app:build-storybook:ci" }
        }
      }
    },
    "tags": [],
    "root": "apps/webpack-app",
    "implicitDependencies": []
  },
  "vite-app": {
    "name": "vite-app",
    "$schema": "../../node_modules/nx/schemas/project-schema.json",
    "sourceRoot": "apps/vite-app/src",
    "projectType": "application",
    "targets": {
      "build": {
        "dependsOn": ["^build"],
        "inputs": ["production", "^production"],
        "executor": "@nx/vite:build",
        "outputs": ["{options.outputPath}"],
        "defaultConfiguration": "production",
        "options": { "outputPath": "dist/apps/vite-app", "mode": "production" },
        "configurations": {
          "development": { "mode": "development" },
          "production": { "mode": "production" }
        }
      },
      "serve": {
        "executor": "@nx/vite:dev-server",
        "defaultConfiguration": "development",
        "options": { "buildTarget": "vite-app:build" },
        "configurations": {
          "development": {
            "buildTarget": "vite-app:build:development",
            "hmr": true
          },
          "production": {
            "buildTarget": "vite-app:build:production",
            "hmr": false
          }
        }
      },
      "preview": {
        "executor": "@nx/vite:preview-server",
        "defaultConfiguration": "development",
        "options": { "buildTarget": "vite-app:build" },
        "configurations": {
          "development": { "buildTarget": "vite-app:build:development" },
          "production": { "buildTarget": "vite-app:build:production" }
        }
      },
      "test": {
        "inputs": ["default", "^production", "{workspaceRoot}/jest.preset.js"],
        "executor": "@nx/vite:test",
        "outputs": ["coverage/apps/vite-app"],
        "options": {
          "passWithNoTests": true,
          "reportsDirectory": "../../coverage/apps/vite-app"
        },
        "configurations": {}
      },
      "lint": {
        "inputs": [
          "default",
          "{workspaceRoot}/.eslintrc.json",
          "{workspaceRoot}/.eslintignore"
        ],
        "executor": "@nx/linter:eslint",
        "outputs": ["{options.outputFile}"],
        "options": {
          "lintFilePatterns": ["apps/vite-app/**/*.{ts,tsx,js,jsx}"]
        },
        "configurations": {}
      },
      "serve-static": {
        "executor": "@nx/web:file-server",
        "options": { "buildTarget": "vite-app:build" }
      },
      "storybook": {
        "executor": "@nx/storybook:storybook",
        "options": { "port": 4400, "configDir": "apps/vite-app/.storybook" },
        "configurations": { "ci": { "quiet": true } }
      },
      "build-storybook": {
        "inputs": ["default", "^production", "!{projectRoot}/.storybook/**/*"],
        "executor": "@nx/storybook:build",
        "outputs": ["{options.outputDir}"],
        "options": {
          "outputDir": "dist/storybook/vite-app",
          "configDir": "apps/vite-app/.storybook"
        },
        "configurations": { "ci": { "quiet": true } }
      },
      "static-storybook": {
        "executor": "@nx/web:file-server",
        "options": {
          "buildTarget": "vite-app:build-storybook",
          "staticFilePath": "dist/storybook/vite-app"
        },
        "configurations": {
          "ci": { "buildTarget": "vite-app:build-storybook:ci" }
        }
      }
    },
    "tags": [],
    "root": "apps/vite-app",
    "implicitDependencies": []
  }
}
