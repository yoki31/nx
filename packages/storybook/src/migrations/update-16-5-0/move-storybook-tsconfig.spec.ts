import {
  NxJsonConfiguration,
  ProjectConfiguration,
  Tree,
  addProjectConfiguration,
  updateJson,
  readNxJson,
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import * as variousProjects from './test-configs/storybook-projects.json';
import moveStorybookTsconfig from './move-storybook-tsconfig';

describe('Ignore @nx/react/plugins/storybook in Storybook eslint plugin', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    updateJson<NxJsonConfiguration>(tree, 'nx.json', (json) => {
      json.namedInputs = {
        production: ['default'],
      };
      return json;
    });
    setupProjects(tree);
    await moveStorybookTsconfig(tree);
  });

  it('should update projects tsconfig.json', async () => {
    expect(
      tree.read(`apps/webpack-app/tsconfig.json`).toString()
    ).toMatchSnapshot();
    expect(
      tree.read(`apps/vite-app/tsconfig.json`).toString()
    ).toMatchSnapshot();
    expect(
      tree.read(`apps/my/custom/place/nested-app/tsconfig.json`).toString()
    ).toMatchSnapshot();
  });

  it('should update Storybook tsconfig.json', async () => {
    expect(
      tree.read(`apps/webpack-app/tsconfig.storybook.json`).toString()
    ).toMatchSnapshot();
    expect(
      tree.read(`apps/vite-app/tsconfig.storybook.json`).toString()
    ).toMatchSnapshot();
    expect(
      tree
        .read(`apps/my/custom/place/nested-app/tsconfig.storybook.json`)
        .toString()
    ).toMatchSnapshot();
  });

  it('should delete old Storybook tsconfig.json', async () => {
    expect(tree.exists(`apps/vite-app/.storybook/tsconfig.json`)).toBeFalsy();
    expect(
      tree.exists(`apps/webpack-app/.storybook/tsconfig.json`)
    ).toBeFalsy();
    expect(
      tree.exists(`apps/my/custom/place/nested-app/.storybook/tsconfig.json`)
    ).toBeFalsy();
  });

  it('should update nx.json inputs', () => {
    expect(readNxJson(tree)).toMatchSnapshot();
  });
});

function setupProjects(tree: Tree) {
  for (const [name, project] of Object.entries(variousProjects)) {
    addProjectConfiguration(tree, name, project as ProjectConfiguration);
  }
  tree.write(
    `apps/my/custom/place/nested-app/tsconfig.json`,
    JSON.stringify({
      compilerOptions: {
        jsx: 'react-jsx',
        allowJs: false,
        esModuleInterop: false,
        allowSyntheticDefaultImports: true,
        strict: true,
        types: ['vite/client', 'vitest'],
      },
      files: [],
      include: [],
      references: [
        {
          path: './tsconfig.app.json',
        },
        {
          path: './tsconfig.spec.json',
        },
        {
          path: './.storybook/tsconfig.json',
        },
      ],
      extends: '../../../../../tsconfig.base.json',
    })
  );
  tree.write(
    `apps/webpack-app/tsconfig.json`,
    JSON.stringify({
      compilerOptions: {
        jsx: 'react-jsx',
        allowJs: false,
        esModuleInterop: false,
        allowSyntheticDefaultImports: true,
        strict: true,
      },
      files: [],
      include: [],
      references: [
        {
          path: './tsconfig.app.json',
        },
        {
          path: './tsconfig.spec.json',
        },
        {
          path: './.storybook/tsconfig.json',
        },
      ],
      extends: '../../tsconfig.base.json',
    })
  );
  tree.write(
    'apps/vite-app/tsconfig.json',
    JSON.stringify({
      compilerOptions: {
        jsx: 'react-jsx',
        allowJs: false,
        esModuleInterop: false,
        allowSyntheticDefaultImports: true,
        strict: true,
        types: ['vite/client', 'vitest'],
      },
      files: [],
      include: [],
      references: [
        {
          path: './tsconfig.app.json',
        },
        {
          path: './tsconfig.spec.json',
        },
        {
          path: './.storybook/tsconfig.json',
        },
      ],
      extends: '../../tsconfig.base.json',
    })
  );

  tree.write(
    'apps/my/custom/place/nested-app/.storybook/tsconfig.json',
    JSON.stringify({
      extends: '../tsconfig.json',
      compilerOptions: {
        emitDecoratorMetadata: true,
        outDir: '',
      },
      files: [
        '../../../../../../node_modules/@nx/react/typings/styled-jsx.d.ts',
        '../../../../../../node_modules/@nx/react/typings/cssmodule.d.ts',
        '../../../../../../node_modules/@nx/react/typings/image.d.ts',
      ],
      exclude: [
        '../**/*.spec.ts',
        '../**/*.spec.js',
        '../**/*.spec.tsx',
        '../**/*.spec.jsx',
      ],
      include: [
        '../src/**/*.stories.ts',
        '../src/**/*.stories.js',
        '../src/**/*.stories.jsx',
        '../src/**/*.stories.tsx',
        '../src/**/*.stories.mdx',
        '*.js',
      ],
    })
  );

  tree.write(
    'apps/vite-app/.storybook/tsconfig.json',
    JSON.stringify({
      extends: '../tsconfig.json',
      compilerOptions: {
        emitDecoratorMetadata: true,
        outDir: '',
      },
      files: [
        '../../../node_modules/@nx/react/typings/styled-jsx.d.ts',
        '../../../node_modules/@nx/react/typings/cssmodule.d.ts',
        '../../../node_modules/@nx/react/typings/image.d.ts',
      ],
      exclude: [
        '../**/*.spec.ts',
        '../**/*.spec.js',
        '../**/*.spec.tsx',
        '../**/*.spec.jsx',
      ],
      include: [
        '../src/**/*.stories.ts',
        '../src/**/*.stories.js',
        '../src/**/*.stories.jsx',
        '../src/**/*.stories.tsx',
        '../src/**/*.stories.mdx',
        '*.js',
      ],
    })
  );

  tree.write(
    'apps/webpack-app/.storybook/tsconfig.json',
    JSON.stringify({
      extends: '../tsconfig.json',
      compilerOptions: {
        emitDecoratorMetadata: true,
        outDir: '',
      },
      files: [
        '../../../node_modules/@nx/react/typings/styled-jsx.d.ts',
        '../../../node_modules/@nx/react/typings/cssmodule.d.ts',
        '../../../node_modules/@nx/react/typings/image.d.ts',
      ],
      exclude: [
        '../**/*.spec.ts',
        '../**/*.spec.js',
        '../**/*.spec.tsx',
        '../**/*.spec.jsx',
      ],
      include: [
        '../src/**/*.stories.ts',
        '../src/**/*.stories.js',
        '../src/**/*.stories.jsx',
        '../src/**/*.stories.tsx',
        '../src/**/*.stories.mdx',
        '*.js',
      ],
    })
  );
}
