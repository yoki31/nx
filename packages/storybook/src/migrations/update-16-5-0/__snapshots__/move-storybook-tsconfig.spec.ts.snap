// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ignore @nx/react/plugins/storybook in Storybook eslint plugin should update Storybook tsconfig.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "emitDecoratorMetadata": true,
    "outDir": ""
  },
  "files": [
    "../../node_modules/@nx/react/typings/styled-jsx.d.ts",
    "../../node_modules/@nx/react/typings/cssmodule.d.ts",
    "../../node_modules/@nx/react/typings/image.d.ts"
  ],
  "exclude": [
    "src/**/*.spec.ts",
    "src/**/*.spec.js",
    "src/**/*.spec.tsx",
    "src/**/*.spec.jsx"
  ],
  "include": [
    "src/**/*.stories.ts",
    "src/**/*.stories.js",
    "src/**/*.stories.jsx",
    "src/**/*.stories.tsx",
    "src/**/*.stories.mdx",
    ".storybook/*.js"
  ]
}
"
`;

exports[`Ignore @nx/react/plugins/storybook in Storybook eslint plugin should update Storybook tsconfig.json 2`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "emitDecoratorMetadata": true,
    "outDir": ""
  },
  "files": [
    "../../node_modules/@nx/react/typings/styled-jsx.d.ts",
    "../../node_modules/@nx/react/typings/cssmodule.d.ts",
    "../../node_modules/@nx/react/typings/image.d.ts"
  ],
  "exclude": [
    "src/**/*.spec.ts",
    "src/**/*.spec.js",
    "src/**/*.spec.tsx",
    "src/**/*.spec.jsx"
  ],
  "include": [
    "src/**/*.stories.ts",
    "src/**/*.stories.js",
    "src/**/*.stories.jsx",
    "src/**/*.stories.tsx",
    "src/**/*.stories.mdx",
    ".storybook/*.js"
  ]
}
"
`;

exports[`Ignore @nx/react/plugins/storybook in Storybook eslint plugin should update Storybook tsconfig.json 3`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "emitDecoratorMetadata": true,
    "outDir": ""
  },
  "files": [
    "../../../../../node_modules/@nx/react/typings/styled-jsx.d.ts",
    "../../../../../node_modules/@nx/react/typings/cssmodule.d.ts",
    "../../../../../node_modules/@nx/react/typings/image.d.ts"
  ],
  "exclude": [
    "src/**/*.spec.ts",
    "src/**/*.spec.js",
    "src/**/*.spec.tsx",
    "src/**/*.spec.jsx"
  ],
  "include": [
    "src/**/*.stories.ts",
    "src/**/*.stories.js",
    "src/**/*.stories.jsx",
    "src/**/*.stories.tsx",
    "src/**/*.stories.mdx",
    ".storybook/*.js"
  ]
}
"
`;

exports[`Ignore @nx/react/plugins/storybook in Storybook eslint plugin should update nx.json inputs 1`] = `
{
  "affected": {
    "defaultBase": "main",
  },
  "namedInputs": {
    "production": [
      "default",
      "!{projectRoot}/**/*.stories.@(js|jsx|ts|tsx|mdx)",
      "!{projectRoot}/.storybook/**/*",
      "!{projectRoot}/tsconfig.storybook.json",
    ],
  },
  "targetDefaults": {
    "build": {
      "cache": true,
    },
    "build-storybook": {
      "inputs": [
        "default",
        "^production",
        "{projectRoot}/.storybook/**/*",
        "{projectRoot}/tsconfig.storybook.json",
      ],
    },
    "lint": {
      "cache": true,
    },
  },
}
`;

exports[`Ignore @nx/react/plugins/storybook in Storybook eslint plugin should update projects tsconfig.json 1`] = `
"{
  "compilerOptions": {
    "jsx": "react-jsx",
    "allowJs": false,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    },
    {
      "path": "./tsconfig.spec.json"
    },
    {
      "path": "./tsconfig.storybook.json"
    }
  ],
  "extends": "../../tsconfig.base.json"
}
"
`;

exports[`Ignore @nx/react/plugins/storybook in Storybook eslint plugin should update projects tsconfig.json 2`] = `
"{
  "compilerOptions": {
    "jsx": "react-jsx",
    "allowJs": false,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "types": ["vite/client", "vitest"]
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    },
    {
      "path": "./tsconfig.spec.json"
    },
    {
      "path": "./tsconfig.storybook.json"
    }
  ],
  "extends": "../../tsconfig.base.json"
}
"
`;

exports[`Ignore @nx/react/plugins/storybook in Storybook eslint plugin should update projects tsconfig.json 3`] = `
"{
  "compilerOptions": {
    "jsx": "react-jsx",
    "allowJs": false,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "types": ["vite/client", "vitest"]
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    },
    {
      "path": "./tsconfig.spec.json"
    },
    {
      "path": "./tsconfig.storybook.json"
    }
  ],
  "extends": "../../../../../tsconfig.base.json"
}
"
`;
